automation/release.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+function cleanup_gh_install() {
+    [ -n "${gh_cli_dir}" ] && [ -d "${gh_cli_dir}" ] && rm -rf "${gh_cli_dir:?}/"
+}
+
+function ensure_gh_cli_installed() {
+    if command -V gh; then
+        return
+    fi
+
+    trap 'cleanup_gh_install' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    gh_cli_dir=$(mktemp -d)
+    (
+        cd  "$gh_cli_dir/"
+        curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+        tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    )
+    export PATH="$gh_cli_dir/gh_${GH_CLI_VERSION}_linux_amd64/bin:$PATH"
+    if ! command -V gh; then
+        echo "gh cli not installed successfully"
+        exit 1
+    fi
+    gh config set prompt disabled
+}
+
+function main() {
+
+    DOCKER_TAG="$(git tag --points-at HEAD | head -1)"
+    if [ -z "$DOCKER_TAG" ]; then
+        echo "commit $(git show -s --format=%h) doesn't have a tag, exiting..."
+        exit 0
+    fi
+    export DOCKER_TAG
+    export TRAVIS_TAG="$DOCKER_TAG"
+
+    ensure_gh_cli_installed
+
+    make
+    make build-verify # verify that we set version on the packages built by go (goveralls depends on go-build target)
+    make apidocs
+    make client-python
+    make manifests DOCKER_PREFIX="$DOCKER_PREFIX" DOCKER_TAG="$DOCKER_TAG" # skip getting old CSVs here (no QUAY_REPOSITORY), verification might fail because of stricter rules over time; falls back to latest if not on a tag
+    make olm-verify
+    make prom-rules-verify
+
+    make manifests
+    make build-functests
+
+    # add artifacts to created github release (overwrite existing)
+    gh release upload --clobber "$DOCKER_TAG" _out/cmd/virtctl/virtctl-v* \
```

Comment:
We need to add a login like [it's done here](https://github.com/kubevirt/project-infra/pull/922/files#diff-dcf92eb27ffaecc04af2f7d69f36b0527860208037e248852334d703518d435aR480), otherwise this won't work!

Reply:
This is already done in the previous line, right? Is there anything else missing here?

Reply:
No, there should be everything present now. I've just put the two PRs on hold as they should both get merged in a reasonable short timespan.